en/lessons/basics/basics.md

```diff
@@ -9,269 +9,197 @@ Getting started, basic data types, and basic operations.
 
 ### Floats
 
-In Elixir, floating point numbers require a decimal after at least one digit; they have 64-bit double precision and support `e` for exponent values:
+Haskell supports `Double`, a double-precision floating-point number.
+At least one digit must come before the decimal point and Haskell supports
+scientific `e` notation.
 
-```elixir
-iex> 3.14
+```haskell
+ghci> 3.14
 3.14
-iex> .14
-** (SyntaxError) iex:2: syntax error before: '.'
-iex> 1.0e-10
-1.0e-10
-```
+ghci> .14
 
-### Booleans
-
-Elixir supports `true` and `false` as booleans; everything is truthy except for `false` and `nil`:
-
-```elixir
-iex> true
-true
-iex> false
-false
+<interactive>:2:1: error: parse error on input ‘.’
+ghci> 5e-10
+5.0e-10
+ghci> 2.7e10
+2.7e10
 ```
 
-### Atoms
-
-An atom is a constant whose name is its value.
-If you're familiar with Ruby, these are synonymous with Symbols:
-
-```elixir
-iex> :foo
-:foo
-iex> :foo == :bar
-false
-```
-
-The booleans `true` and `false` are also the atoms `:true` and `:false`, respectively.
-
-```elixir
-iex> is_atom(true)
-true
-iex> is_boolean(:true)
-true
-iex> :true === true
-true
-```
-
-Names of modules in Elixir are also atoms. `MyApp.MyModule` is a valid atom, even if no such module has been declared yet.
-
-```elixir
-iex> is_atom(MyApp.MyModule)
-true
-```
+### Booleans
 
-Atoms are also used to reference modules from Erlang libraries, including built in ones.
+Booleans in Haskell are represented with `True` and `False`:
 
-```elixir
-iex> :crypto.strong_rand_bytes 3
-<<23, 104, 108>>
+```haskell
+ghci> True
+True
+ghci> False
+False
 ```
 
-### Strings
+### Text
 
-Strings in Elixir are UTF-8 encoded and are wrapped in double quotes:
+Haskell supports unicode text. Strings are wrapped in double quotes:
 
-```elixir
-iex> "Hello"
+```haskell
+ghci> "Hello" :: T.Text
 "Hello"
-iex> "dziękuję"
-"dziękuję"
+ghci> "jalapeño" :: T.Text
+"jalape\241o"
 ```
 
-Strings support line breaks and escape sequences:
+Special characters can be represented with numeric escapes or escape codes:
 
-```elixir
-iex> "foo
-...> bar"
-"foo\nbar"
-iex> "foo\nbar"
-"foo\nbar"
+```haskell
+ghci> "'hi\"" :: T.Text
+"'hi\""
+ghci> "\n" :: T.Text
+"\n"
+ghci> "\42" :: T.Text
+"*"
 ```
 
-Elixir also includes more complex data types.
-We'll learn more about these when we learn about [collections](../collections/) and [functions](../functions/).
+You'll learn more about data types in [collections](../collections/) and
+[functions](../functions/).
 
 ## Basic Operations
 
 ### Arithmetic
 
-Elixir supports the basic operators `+`, `-`, `*`, and `/` as you would expect.
-It's important to remember that `/` will always return a float:
+Haskell supports the basic operators `+`, `-`, `*`, and `/`.
+Note that `/` never returns an integer.
 
-```elixir
-iex> 2 + 2
+```haskell
+ghci> 2 + 2
 4
-iex> 2 - 1
+ghci> 2 - 1
 1
-iex> 2 * 5
+ghci> 2 * 5
 10
-iex> 10 / 5
+ghci> 10 / 5
 2.0
 ```
```

Comment:
I'd add `^` and `**` which are surprisingly less known (in my experience) but are a must in today's data-centric world.

Reply:
Agreed